src/components/ProductTours/AdminOnboardingTours/data/useAdminOnboardingTour.tsx

```diff
@@ -0,0 +1,67 @@
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+import { logError } from '@edx/frontend-platform/logging';
+
+import {
+  ANALYTICS_INSIGHTS_TARGETS,
+  ORGANIZE_LEARNER_TARGETS,
+  TRACK_LEARNER_PROGRESS_TARGETS,
+} from '../constants';
+
+import LmsApiService from '../../../../data/services/LmsApiService';
+import useCreateLearnerProgressFlow from './useCreateLearnerProgressFlow';
+import useCreateAnalyticsFlow from './useCreateAnalyticsFlow';
+import { TourStep } from '../../types';
+import useCreateOrganizeLearnersFlow from './useCreateOrganizeLearnersFlow';
+
+interface UseAdminOnboardingTourProps {
+  adminUuid: string;
+  aiButtonVisible: boolean;
+  currentStep: number;
+  setCurrentStep: (currentStep: number) => void;
+  enterpriseSlug: string;
+  targetSelector?: string;
+}
+
+const useAdminOnboardingTour = (
+  {
+    adminUuid, aiButtonVisible, currentStep, setCurrentStep, enterpriseSlug, targetSelector,
+  }: UseAdminOnboardingTourProps,
+): Array<TourStep> => {
+  const handleAdvanceTour = (advanceEventName: string) => {
+    const newIndex = currentStep + 1;
+    sendEnterpriseTrackEvent(enterpriseSlug, advanceEventName, { 'completed-step': newIndex });
+    setCurrentStep(newIndex);
+  };
+
+  const handleEndTour = async (endEventName: string, flowUuid: string) => {
```

Comment:
It might look like this is a hook because it returns functions that are defined here, but since it doesn't use useEffect / react lifecycle, it's a factory function.